src/modules/back-to-top/back-to-top.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
+
+@Component({
+  selector: 'stache-back-to-top',
+  templateUrl: './back-to-top.component.html',
+  styleUrls: ['./back-to-top.component.scss']
+})
+export class StacheBackToTopComponent {
+  @Input() offset: number;
+  @Input() color: string;
+  @Input() backgroundColor: string;
+  @ViewChild('backToTop') el: ElementRef;
+
+  ngOnInit(): void {
+    window.addEventListener('scroll', () => {
+      if (window.pageYOffset > this.offset) {
+        this.el.nativeElement.classList.add('show');
+      } else {
+        this.el.nativeElement.classList.remove('show');
+      }
+    });
+  }
+
+  goToTop() {
+    try {
```

Comment:
If you can think of a way to add a fallback without it I'm open to suggestions. This seemed like the simplest solution to ensure the back to top worked regardless.